clang-tools-extra/clang-tidy/modernize/AvoidBindCheck.cpp

```diff
@@ -554,6 +555,8 @@ getLambdaProperties(const MatchFinder::MatchResult &Result) {
   LP.Callable.Materialization = getCallableMaterialization(Result);
   LP.Callable.Decl =
       getCallMethodDecl(Result, LP.Callable.Type, LP.Callable.Materialization);
+  if (LP.Callable.Decl)
+    LP.Callable.DoesReturn = !LP.Callable.Decl->getReturnType()->isVoidType();
```

Comment:
check here also that return type is not null, just in case..., you may also try to check a canonical type, in case that void would be somehow hidden behind some typedefs/templates or something

Reply:
Maybe: getReturnType().getCanonicalType().getTypePtr()

Reply:
Done. Must have gotten lost